dlib/dnn/loss.h

```diff
@@ -2832,21 +2966,7 @@ namespace dlib
     {
     public:
 
-        struct weighted_label
```

Comment:
Hmm. Now existing code will break, when it tries to instantiate this nested (now-missing) struct. Not saying it's the end of the world (probably not so many people are using this), but on the other hand I think it's relatively easy to maintain compatibility, so perhaps it's worthwhile to at least give it a try?

I think you can leave here something like `typedef dlib::weighted_label<uint16_t> weighted_label;`? You'll know this works, if you can then revert [the changes you had to make to the test program](https://github.com/davisking/dlib/commit/a9033b58d1858a3271bfa6620f420475f5f2c822#diff-7d2445557c03b31a758c2ef7d310bc53).

If my comment is not clear, let me know and I'll see if I can write it as code instead.

Reply:
I think I got your point :)